rest_server/RestServer.go

```diff
@@ -0,0 +1,269 @@
+	if len(config["log.console.destination"]) == 0 {
+		return fmt.Errorf("missing parameter 'log.console.destination'")
+	}
+	rs.logConsoleDestination = config["log.console.destination"]
+	if len(config["log.console.severity"]) == 0 {
+		return fmt.Errorf("missing parameter 'log.console.severity'")
+	}
+	rs.logConsoleSeverity = config["log.console.severity"]
+
+	if rs.Ipv4HTTPPort, err = strconv.ParseUint(config["rest_server.ipv4_http_port"], 10, 64); err != nil {
+		return err
+	}
+	if rs.Ipv4HTTPSPort, err = strconv.ParseUint(config["rest_server.ipv4_https_port"], 10, 64); err != nil {
+		return err
+	}
+	if rs.Ipv6HTTPPort, err = strconv.ParseUint(config["rest_server.ipv6_http_port"], 10, 64); err != nil {
+		return err
+	}
+	if rs.Ipv6HTTPSPort, err = strconv.ParseUint(config["rest_server.ipv6_https_port"], 10, 64); err != nil {
+		return err
+	}
+	if len(config["rest_server.tls_certificate_chain"]) > 0 {
+		if _, err := os.Stat(config["rest_server.tls_certificate_chain"]); os.IsNotExist(err) {
+			if _, err := os.Stat(SiodbInstanceConfigurationPath + `/` + config["rest_server.tls_certificate_chain"]); os.IsNotExist(err) {
+				return err
+			} else {
+				rs.TLSCertificate = SiodbInstanceConfigurationPath + `/` + config["rest_server.tls_certificate_chain"]
+			}
+
+		} else {
+			rs.TLSCertificate = config["rest_server.tls_certificate_chain"]
+		}
+	} else {
+		if _, err := os.Stat(config["rest_server.tls_certificate"]); os.IsNotExist(err) {
+			if _, err := os.Stat(SiodbInstanceConfigurationPath + `/` + config["rest_server.tls_certificate"]); os.IsNotExist(err) {
+				return err
+			} else {
+				rs.TLSCertificate = SiodbInstanceConfigurationPath + `/` + config["rest_server.tls_certificate"]
+			}
+		} else {
+			rs.TLSCertificate = config["rest_server.tls_certificate"]
+		}
+	}
+	if _, err := os.Stat(config["rest_server.tls_private_key"]); os.IsNotExist(err) {
+		if _, err := os.Stat(SiodbInstanceConfigurationPath + `/` + config["rest_server.tls_private_key"]); os.IsNotExist(err) {
+			return err
+		} else {
+			rs.TLSPrivateKey = SiodbInstanceConfigurationPath + `/` + config["rest_server.tls_private_key"]
+		}
+	} else {
+		rs.TLSPrivateKey = config["rest_server.tls_private_key"]
+	}
+
+	// Parse rest_server.chunk_size
+	var Unit string = config["rest_server.chunk_size"][len(config["rest_server.chunk_size"])-1:]
+	var IsLetter = regexp.MustCompile(`^[a-zA-Z]+$`).MatchString
+
+	if IsLetter(Unit) {
+		if rs.HTTPChunkSize, err = strconv.ParseUint(config["rest_server.chunk_size"][:len(config["rest_server.chunk_size"])-1], 10, 64); err != nil {
+			return err
+		}
+		switch Unit {
+		case "k", "K":
+			rs.HTTPChunkSize = rs.HTTPChunkSize * 1024
+		case "m", "M":
+			rs.HTTPChunkSize = rs.HTTPChunkSize * 1024 * 1024
+		default:
+			return fmt.Errorf("unknown unit '%v' for parameter 'rest_server.chunk_size'", Unit)
+		}
+	} else {
+		if rs.HTTPChunkSize, err = strconv.ParseUint(config["rest_server.chunk_size"], 10, 64); err != nil {
```

Comment:
all such buffer sizes must be validated to be nonzero and not exceed some reasonable limit